### PR TITLE
[LoongArch64] Amend the nativeaot for LoongArch64.

### PR DESCRIPTION
**Pull request by sunlijun-610**
This PR fix several mistakes in nativeaot on LoongArch64:
1. Compilation error in debug mode.
2. The order of `m_RIP` and `m_FramePointer` in `struct PInvokeTransitionFrame`.
3. Add the test for invalid entries in `EmitDictionaryLookup`, in `class ReadyToRunGenericHelperNode`.

The changes have been tested and verified to be successful.

### CODE REVIEW

**Review by shushanhf — APPROVED** [review]
LGTM
Thanks

File: src/coreclr/nativeaot/Runtime/inc/rhbinder.h [hunk 1]
```diff
@@ -461,7 +461,7 @@ struct PInvokeTransitionFrame
 #else // USE_PORTABLE_HELPERS
 struct PInvokeTransitionFrame
 {
-#if defined(TARGET_ARM64) || defined(TARGET_ARM)
+#if defined(TARGET_ARM64) || defined(TARGET_ARM) || defined(TARGET_LOONGARCH64)
     // On arm32/arm64, the FP and LR registers are pushed in that order when setting up frames
```
Comment (MichalStrehovsky) [inline src/coreclr/nativeaot/Runtime/inc/rhbinder.h:465]:
```suggestion
#if defined(TARGET_ARM64) || defined(TARGET_ARM) || defined(TARGET_LOONGARCH64)
    // The FP and LR registers are pushed in different order when setting up frames
```

**Review by MichalStrehovsky — APPROVED** [review]

**Comment by am11** [comment]
> The changes have been tested and verified to be successful.

There are three sets of AOT tests: smoke tests, coreclr tests and libraries tests: https://github.com/dotnet/runtime/blob/cd9cc877d157684dd69ccc4d2cabaa60368b1524/docs/workflow/testing/coreclr/testing.md#building-the-tests. Can you open an issue to track their progress; how many of which category are passing and what's remaining?

**Comment by sunlijun-610** [comment]
> Can you open an issue to track their progress; how many of which category are passing and what's remaining?

OK, I've opened an issue to track. #104661 
Afterwards, I will summarize the current testing progress and update it as soon as possible.
Thanks!

**Pull request merged**